src/backend/metal/src/native.rs

```diff
@@ -159,6 +159,12 @@ impl<T> ResourceData<T> {
             samplers: fun(&self.samplers),
         }
     }
+
+    pub fn consume<V, F: FnMut(&mut T, V)>(&mut self, other: ResourceData<V>, mut fun: F) {
```

Comment:
This is always passed the same function, could we just put the logic here instead?